File: admin/src/features/rooms/api/dto/room-dto.ts
import { Nullable } from "@/types/utility";
import { mapNullable } from "@/utils/fp";
import {
  CityDto,
  LangRecordDto,
  fromCityDto,
  fromLangRecordDto,
} from "@/features/misc";
import { ImageDto, fromImageDto } from "@/features/images";
import { Room } from "@/features/rooms";

export interface RoomDto {
  id: string;
  titles: LangRecordDto<string>;
  descriptions: LangRecordDto<string>;
  peculiarities: LangRecordDto<string[]>;
  price: number;
  city: CityDto;
  cover: Nullable<ImageDto>;
  images: ImageDto[];
}

export const fromRoomDto = (dto: RoomDto): Room => ({
  id: dto.id,
  titles: fromLangRecordDto(dto.titles),
  descriptions: fromLangRecordDto(dto.descriptions),
  peculiarities: fromLangRecordDto(dto.peculiarities),
  price: dto.price,
  city: fromCityDto(dto.city),
  cover: mapNullable(fromImageDto)(dto.cover),
  images: dto.images.map(fromImageDto),
});

File: backend/src/Hotel.Orbital.Core/Models/RoomDto.cs
using System.ComponentModel.DataAnnotations;
using Entities.Enums;

namespace Core.Models;

/// <summary>
/// Dto номера гостиницы
/// </summary>
public class RoomDto
{
    /// <summary>
    /// Идентфиикатор
    /// </summary>
    [Required]
    public Guid Id { get; set; }
    
    /// <summary>
    /// Город
    /// </summary>
    [Required]
    public City City { get; set; }

    /// <summary>
    /// Заголовок
    /// </summary>
    [Required]
    public Dictionary<Language, string> Titles { get; set; }

    /// <summary>
    /// Описание
    /// </summary>
    [Required]
    public Dictionary<Language, string> Descriptions { get; set; }

    /// <summary>
    /// Особенности
    /// </summary>
    [Required]
    public Dictionary<Language, IEnumerable<string>> Peculiarities { get; set; }

    /// <summary>
    /// Цена
    /// </summary>
    [Required]
    public float Price { get; set; }

    /// <summary>
    /// Идентификатор обложки
    /// </summary>
    public ImageDto Cover { get; set; }

    /// <summary>
    /// Идентификаторы изображений номера
    /// </summary>
    public List<ImageDto> Images { get; set; }
}